e2e/specs/drug-orders.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { expect } from '@playwright/test';
+import { type Visit } from '@openmrs/esm-framework';
+import { generateRandomPatient, deletePatient, type Patient, startVisit, endVisit } from '../commands';
+import { test } from '../core';
+import { MedicationsPage } from '../pages';
+
+let patient: Patient;
+let visit: Visit;
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+  visit = await startVisit(api, patient.uuid);
+});
+
+test('Record, edit and discontinue a drug order', async ({ page }) => {
+  const medicationsPage = new MedicationsPage(page);
+
+  await test.step('When I visit the medications page', async () => {
+    await medicationsPage.goTo(patient.uuid);
+  });
+
+  await test.step('And I click on the `Record active medications` link', async () => {
+    await medicationsPage.page.getByText(/record active medications/i).click();
+  });
+
+  await test.step('And I fill the drug order form', async () => {
+    await expect(medicationsPage.page.getByText(/add drug order/i)).toBeVisible();
```

Comment:
Avoid adding assertions within when blocks. Instead, place them within then blocks for clearer and more organized testing

Reply:
Check https://github.com/openmrs/openmrs-esm-patient-chart/pull/1714#discussion_r1512573196